### PR TITLE
5.5 different floor nav

### PR DESCRIPTION
**Pull request by vtaing10**
## Description:
Indoor Navigation between 2 separate floors

## Related User Story:
5.4 (indoors direction)

## Screenshot/Demo:
image
image

## How to test:
Select 2 separate rooms in the same building no matter the floor. You should get directions and a "next step" button that guides you through the directions.
## Type of Change:
- [x] Feature
- [ ] Bug fix
- [ ] Refactor
- [ ] Documentation update
- [ ] Tests
- [ ] Others

## Pull Request Checklist
- [ ] Flutter CI test passed
- [ ] Sonar test passed
- [ ] Codecov test passed
- [ ] Application runs without errors
- [ ] Feature tested locally on a high level
- [ ] Architecture conventions respected
- [ ] No API keys or secrets committed
- [ ] Document when necessary

## PR Review Checklist (dev)
- [ ] Code is readable and well structured
- [ ] No obvious bugs or edge cases missed
- [ ] Tests are adequate
- [ ] Naming and organization are clear
- [ ] Feature works as described
- [ ] No unnecessary complexity
- [ ] Follows project architecture

## PR Review Checklist (QA)
- [ ] Made sure to test the application fully
- [ ] No bugs created by this PR

### CODE REVIEW

File: lib/services/indoor_maps/indoor_map_repository.dart [hunk 1]
```diff
@@ -199,4 +232,21 @@ class IndoorMapRepository {
 
     return null;
   }
+
+  String? _extractFloorLevel(List<dynamic> features) {
```
Comment (maddie-luu) [inline lib/services/indoor_maps/indoor_map_repository.dart]: So this method is concise and easy to understand. It scans features and returns the firt valid level. I do like the use of ? and .toString() to avoid immediate null crashes in addition to early return optimization where it stops as soon as a valid leve is found, demonstrating efficiency.

File: lib/services/indoor_maps/indoor_map_repository.dart [hunk 2]
```diff
@@ -199,4 +232,21 @@ class IndoorMapRepository {
...
+    for (final feature in features) {
+      try {
+        final featureMap = feature as Map<String, dynamic>;
```
Comment (maddie-luu) [inline lib/services/indoor_maps/indoor_map_repository.dart]: There's some unsafe casting here, this will throw if feature isn't the expected type. Use instead safe type checks (is) instead of `as`.

File: lib/services/indoor_maps/indoor_map_repository.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (vtaing10) [reply lib/services/indoor_maps/indoor_map_repository.dart]: Makes sense. I kept the broader input type for compatibility with the GeoJSON decode path, but added element validation before access so we still get safer behavior without forcing a wider refactor.

File: lib/services/indoor_maps/indoor_map_repository.dart [hunk 3]
```diff
@@ -199,4 +232,21 @@ class IndoorMapRepository {
...
+        final props = featureMap['properties'] as Map<String, dynamic>?;
+        final level = props?['level']?.toString();
+        if (level != null && level.trim().isNotEmpty) {
+          return level;
+        }
+      } catch (_) {
+        continue;
+      }
+    }
```
[the remark this answers is not in the document]
Comment (vtaing10) [reply lib/services/indoor_maps/indoor_map_repository.dart]: Agreed. I added debug-only logging for malformed features so bad data is easier to trace without adding production noise.

File: lib/services/indoors_routing/indoor_multi_floor_router.dart [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+  final IndoorSameFloorRouter sameFloorRouter;
+  final IndoorGeometry geometry;
+  final IndoorTransitionMatcher transitionMatcher;
+
+  IndoorMultiFloorRouter({
+    IndoorSameFloorRouter? sameFloorRouter,
+    IndoorGeometry? geometry,
+    IndoorTransitionMatcher? transitionMatcher,
+  }) : sameFloorRouter = sameFloorRouter ?? IndoorSameFloorRouter(),
+       geometry = geometry ?? const IndoorGeometry(),
```
[the remark this answers is not in the document]
Comment (vtaing10) [reply lib/services/indoors_routing/indoor_multi_floor_router.dart]: Agreed. I refactored the constructor so geometry is resolved once and reused for both the router field and the default transition matcher.

**Review by enyc24 — APPROVED** [review]
